sup3r/postprocessing/collection.py

```diff
@@ -578,6 +571,51 @@ def collect_flist(self, file_paths, out_file, feature, sort=False,
             logger.warning(msg)
             warn(msg)
 
+    def collect_flist(self, file_paths, out_file, feature, sort=False,
+                      sort_key=None, max_workers=None,
+                      target_final_meta_file=None, masked_target_meta=None):
+        """Collect a dataset from a file list with data pre-init.
+
+        Collects data that can be chunked in both space and time.
+
+        Parameters
+        ----------
+        file_paths : list | str
+            Explicit list of str file paths that will be sorted and collected
+            or a single string with unix-style /search/patt*ern.h5. Files
+            should have non-overlapping time_index dataset and fully
+            overlapping meta dataset.
+        collect_dir : str
+            Directory of chunked files (flist).
+        out_file : str
+            File path of final output file.
+        dset : str
+            Dataset name to collect.
+        sort : bool
+            flag to sort flist to determine meta data order.
+        sort_key : None | fun
+            Optional sort key to sort flist by (determines how meta is built
+            if out_file does not exist).
+        max_workers : int | None
+            Number of workers to use in parallel. 1 runs serial,
+            None uses all available.
+        target_final_meta_file : str
+            Path to target final meta containing coordinates to keep from the
+            full file list collected meta
+        masked_target_meta : pd.DataFrame
+            Collected meta data with mask applied from target final meta so
+            original gids are preserved.
```

Comment:
its weird that you need both of these meta arguments here. Why is that? Is it just to improve performance somehow?

Honestly you should just remove the public `collect_flist()`, it's only called in a single place and only modifies the protected `_collect_flist()` with a single additional function call that could be easily moved to the place `collect_flist()` is called. I'd recommend only keeping the protected method because the arguments are confusing and I think are meant to be called indirectly.

Reply:
Yeah it's to improve performance, since we only have to mask the full meta once. Good suggestion on removing this though